protocol/messenger_curated_communities.go

```diff
@@ -0,0 +1,147 @@
+package protocol
+
+import (
+	"context"
+	"errors"
+	"reflect"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/status-im/status-go/eth-node/types"
+	"github.com/status-im/status-go/protocol/communities"
+)
+
+const (
+	curatedCommunitiesUpdateInterval = 2 * time.Minute
+)
+
+// Regularly gets list of curated communities and signals them to client
+func (m *Messenger) startCuratedCommunitiesUpdateLoop() {
+	logger := m.logger.Named("startCuratedCommunitiesUpdateLoop")
+
+	go func() {
+		// Initialize interval to 0 for immediate execution
+		var interval time.Duration = 0
+
+		cache, err := m.communitiesManager.GetCuratedCommunities()
+		if err != nil {
+			logger.Error("failed to start curated communities loop", zap.Error(err))
+			return
+		}
+
+		for {
+			select {
+			case <-time.After(interval):
+				// Immediate execution on first run, then set to regular interval
+				interval = curatedCommunitiesUpdateInterval
+
+				curatedCommunities, err := m.getCuratedCommunitiesFromContract()
+				if err != nil {
+					logger.Error("failed to get curated communities from contract", zap.Error(err))
+					continue
+				}
+
+				if reflect.DeepEqual(cache.ContractCommunities, curatedCommunities.ContractCommunities) &&
+					reflect.DeepEqual(cache.ContractFeaturedCommunities, curatedCommunities.ContractFeaturedCommunities) {
+					// nothing changed
+					continue
+				}
+
+				err = m.communitiesManager.SetCuratedCommunities(curatedCommunities)
+				if err == nil {
+					cache = curatedCommunities
+				} else {
+					logger.Error("failed to save curated communities", zap.Error(err))
+				}
+
+				response, err := m.fetchCuratedCommunities(curatedCommunities)
```

Comment:
Can we just pass `curatedCommunities` to `SendCuratedCommunitiesUpdate`. Maybe we can just wrap it in the Response?

Reply:
Not sure I am following :thinking:  The response is a type crafted specifically for clients, including fetched community data, not only the IDs of curated and featured communities.